src/core/structures/PDFAcroForm.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  PDFArray,
+  PDFBool,
+  PDFContext,
+  PDFDict,
+  PDFName,
+  PDFNumber,
+  PDFStream,
+  PDFString,
+  XFAResource,
+} from 'src/core';
+import { DictMap } from 'src/core/objects/PDFDict';
+
+class PDFAcroForm extends PDFDict {
+  static fromMapWithContext(
+    dict: DictMap,
+    context: PDFContext,
+  ): PDFAcroForm {
+    if (!dict.has(PDFName.of('Fields'))) {
+      dict.set(PDFName.of('Fields'), context.obj([]));
+    }
+    return new PDFAcroForm(dict, context);
+  }
+
+  protected constructor(map: DictMap, context: PDFContext) {
+    super(map, context);
+  }
+
+  Fields(): PDFArray {
+    const fieldRefs = this.lookup(PDFName.of('Fields'), PDFArray);
+    return fieldRefs.map((fieldRef) => this.context.lookup(fieldRef, PDFDict));
+  }
```

Comment:
All instances of `PDFObject` subclasses should exist as values in the `PDFContext` map or else be referenced by another `PDFObject` that is present in the `PDFContext`. Another way of saying this is that no `PDFObject` instances should be generated that are not going to be serialized when `PDFDocument.save()` is called. This is because `PDFObject` subclasses only exist to model objects in a PDF file. 

For example, it would be very easy for somebody to call `PDFAcroForm.Fields` and think they are getting the actual `Fields` array, since a `PDFArray` object is returned. And then they might modify that `PDFArray` thinking they are adding new fields that will be serialized when they save the document. But of course, this would not happen since the array they modified wasn't actually part of the `PDFDocument`.

The map being performed here violates this convention. The `PDFArray` returned by `PDFAcroForm.Fields` is an "orphan". It will not be serialized when `PDFDocument.save()` is invoked. This means that there's no reason for it to be a `PDFArray` in the first place. I think it would be best to just return a `PDFDict[]` instead.

Of course, if this method wasn't mapping and it just returned the value of the `/Fields` entry, it would make sense to return a `PDFArray`. Because it would be returning an object that actually exists in the PDF file.

An alternative to returning `PDFDict[]` here would be to create two methods, for example:
```js
Fields(): PDFArray {
  return this.lookup(PDFName.of('Fields'), PDFArray);
}

get fieldDicts(): PDFDict[] {
  const fieldRefs = this.Fields();
  const fieldDicts = new Array(fieldRefs.size());
  for (let idx = 0, len = fieldRefs.size(); idx < len; idx++) {
    fieldDicts[idx] = this.context.lookup(fieldRefs[idx], PDFDict);
  }
  return fieldDicts;
}
```

Reply:
Updated